src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java

```diff
@@ -295,24 +318,65 @@ public void testCacheRetainingValuesFromTwoConcurrentCalls() throws Exception {
         }
     }
 
-    private static CachingHttpClientBuilder createClientBuilder(Supplier<String> contentSupplier) {
-        // mock client builder
-        CachingHttpClientBuilder clientBuilder = CachingHttpClientBuilder.create();
-        clientBuilder.setConnectionManager(new BasicHttpClientConnectionManager() {
-            @Override
-            public void connect(HttpClientConnection conn, HttpRoute route, int connectTimeout, HttpContext context) {
-            }
-        });
-        clientBuilder.setRequestExecutor(new HttpRequestExecutor() {
-            @Override
-            protected HttpResponse doSendRequest(HttpRequest request, HttpClientConnection conn, HttpContext context)
-                    throws IOException {
-                HttpResponse response = new BasicHttpResponse(HttpVersion.HTTP_1_1, HttpStatus.SC_OK, "Ok");
-                response.setEntity(new StringEntity(contentSupplier.get() + "\n"));
-                return response;
-            }
-        });
+    /**
+     * Plugin execution should fail if a non-200 code was returned by the resource being downloaded.
+     */
+    @Test
+    public void testBuildShouldFailIfDownloadFails() {
+        this.wireMock.stubFor(get(anyUrl()).willReturn(forbidden()));
+        try {
+            createMojo(m -> {
+                setVariableValueToObject(m, "uri", URI.create(wireMock.baseUrl()));
+                setVariableValueToObject(m, "skipCache", true);
+                setVariableValueToObject(m, "failOnError", true);
+            }).execute();
+            fail("The mojo should have failed upon error");
+        } catch (Exception e) {
+            assertThat(e, is(instanceOf(MojoExecutionException.class)));
+            assertThat(e.getCause(), is(instanceOf(DownloadFailureException.class)));
+            assertThat(((DownloadFailureException) e.getCause()).getHttpCode(), is(HttpStatus.SC_FORBIDDEN));
+        }
```

Comment:
@ajarmoniuk let's check the actual number of retries happened here

Reply:
Done.